src/helpers.js

```diff
@@ -141,4 +155,16 @@ module.exports = {
         const requests = createRequests(requestOptions);
         await addRequestsToQueue({ requests, requestQueue, startUrl, maxRequestsPerStartUrl, requestsPerStartUrlCounter });
     },
+
+    normalizeUrls: (urls) => {
+        const URL_PREFIX = 'http://';
+        return urls.map(({ url }) => {
```

Comment:
this can be changed with `const nUrl = new URL(url)` and mutating the property on it, like `nUrl.protocol = 'http:';` then calling `nUrl.toString()` at the end. 
it will have the proper URL without needing to use replace or checking for string existence

Reply:
The issue requires support of all of the following url formats:

- http://www.apify.com
- www.apify.com
- [apify.com](http://www.apify.com)

`const nUrl = new URL(url)` fails for both `url === 'www.apify.com'` and `url === 'apify.com'`. 

It could be used with relative urls but `//` would have to be prepended to all urls without the protocol specified. These are valid calls creating `'http://apify.com/'` and `'http://www.apify.com/'` respectively:

```javascript
const firstUrl = new URL("//apify.com", "http://example.com")
const secondUrl = new URL("//www.apify.com", "http://example.com")
```

The function could then be transformed to:
```javascript
normalizeUrls: (urls) => {
        const PROTOCOL_REGEX = /^((.)+:\/\/)/;
        const BASE_URL_PATTERN = 'http://example.com';

        return urls.map(({ url }) => {
            const urlWithoutProtocol = url.replace(PROTOCOL_REGEX, '');
            const relativeUrl = `//${urlWithoutProtocol}`;
            const normalizedUrl = new URL(relativeUrl, BASE_URL_PATTERN)
    
            return normalizedUrl.toString();
        });
    }
```